### ISSUE TITLE
How to retrieve current SizeRollingFileTree?

### ISSUE THREAD

**Issue by sceee** [magillus/flutter-fimber#47]
I am using SizeRollingFileTree and I want to retrieve the current log file from the filesystem so I can display it etc.
How can I achieve this? Is there a function that gives me the current log or do I have to search the filesystem by myself?

**Comment by magillus** [comment]
Hi,
You can store the instance of SizeRollingFileTree in a variable to access it.
After that you can read property `outputFileName` that is current filename used.

**Comment by sceee** [comment]
Hi @magillus,

Thanks, good hint.
But now I am running in the problem that the outputFileName is always `.`.

Also when I try to log something, the follwing exception is thrown:
`E/flutter (18939): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: FileSystemException: Cannot open file, path = '.' (OS Error: Is a directory, errno = 21)`

This is how I initialize Fimber:
```
    String logFilePrefix = '/data/user/0/com.my.app/app_flutter/log_';

    _rollingFile = SizeRollingFileTree(
      DataSize.mega(3),
      logLevels: CustomFormatTree.defaultLevels,
      filenamePrefix: logFilePrefix,
    );

    Fimber.plantTree(DebugTree());
    Fimber.plantTree(_rollingFile);
```

When I then try to log something via `Fimber.i(...)`, the mentioned exception is thrown.
After having a look at `RollingFileTree`, the SizeRollingFileTree initializes its super super class `FimberFileTree` with "." (RollingFileTree constructor calling its super class).

Am I doing something wrong?

**Comment by magillus** [comment]
hmm.. thank you for details, I will try this on a Flutter project and I will take a look this at first convenient time.

**Comment by magillus** [comment]
I tried this on example project and it worked fine - I used emulator.

Check if path is correct for the project's directory via `adb shell` 

Here is output from couple shell commands:

```
generic_x86:/data/user/0/com.perlak.flutterfimberexample/app_flutter # ls -l
total 16
drwx------ 2 u0_a88 u0_a88 4096 2019-07-01 06:12 flutter_assets
-rw------- 1 u0_a88 u0_a88  197 2019-07-01 06:13 log_1.txt
-rw------- 1 u0_a88 u0_a88    0 2019-07-01 06:12 res_timestamp-1-1561979535986
```
content of the log file.
```
generic_x86:/data/user/0/com.perlak.flutterfimberexample/app_flutter # cat log_1.txt
2019-07-01T06:13:28.173360      I _MyAppState.build.<ac>: Info message test 2019-07-01 06:13:28.150136
2019-07-01T06:13:29.405237      E _MyAppState.build.<ac>: Error message test 2019-07-01 06:13:29.391612

```

Make sure that iOS device also is supported by the given path - maybe use `https://pub.dev/packages/path_provider` package for giving the paths

**Comment by sceee** [comment]
@magillus I tried it with path_provider but unfortunately I can't get it to work. I receive the same error as mentioned above.

I guess it's because in the `RollingFileTree` constructor, the super constructor (`FimberFileTree`) is called with the static file name `"."`.

**Comment by sceee** [comment]
Ok, I think I found at least one issue. I am not sure if this causes the complete issue.
In the `SizeRollingFileTree` constructor, the async method `detectFileIndex()` is called.
Obviously, the constructor immediately returns (since this call is not `await` which would anyway not be possible since constructors can't be `async`).

This causes `_flushBuffer(..)` to be called with an `outputFileName` of `.` - so the default. Obviously this throws an exception.

**Comment by magillus** [comment]
That is great investigation, thank you. Will take a look at PR.

**Comment by magillus** [comment]
fixed in `0.3.1` - published

**Comment by magillus** [comment]
ups, in 0.3.2 published - the publish 0.3.1 didn't succeed

**Comment by sceee** [comment]
Thanks.
